java/test/jmri/jmrit/display/controlPanelEditor/shape/DrawCircleTest.java

```diff
@@ -1,58 +1,61 @@
 package jmri.jmrit.display.controlPanelEditor.shape;
 
-import java.awt.GraphicsEnvironment;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 import jmri.jmrit.display.EditorScaffold;
-import jmri.jmrit.display.controlPanelEditor.ControlPanelEditor;
 import jmri.util.JUnitUtil;
+import jmri.util.ThreadingUtil;
+import jmri.util.junit.annotations.DisabledIfHeadless;
 
 import org.junit.jupiter.api.*;
-import org.junit.Assert;
-import org.junit.Assume;
 
 /**
  *
  * @author Paul Bender Copyright (C) 2017
  */
+@DisabledIfHeadless
 public class DrawCircleTest {
 
-    EditorScaffold editor;
+    private EditorScaffold editor;
 
     @Test
     public void testCTor() {
-        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
-        ControlPanelEditor frame = new ControlPanelEditor();
-        frame.pack();
-        frame.setVisible(true);
-        DrawCircle t = new DrawCircle("newShape", "Circle", null, frame, false);
-        Assert.assertNotNull("exists", t);
+
+        ThreadingUtil.runOnGUI( () -> {
+            editor.pack();
+            editor.setVisible(true);
+        });
+        DrawCircle t = new DrawCircle("newShape", "Circle", null, editor, false);
```

Comment:
Doesn't the call to DrawCircle() have any action that needs to be on the swing thread? Or does any drawing of the circle happens later? It seems to me that this call should also be within the runOnGUI() call. But I might be wrong.